commons/Common.mli

```diff
@@ -140,6 +140,44 @@ val write_file : file:filename -> string -> unit
 *)
 val read_file : ?max_len:int -> filename -> string
 
+val is_directory : filename -> bool
+
+
+type path_special = private Path of string [@@deriving show, eq]
+
+module Path :
```

Comment:
I would put it in a separate file, commons/Path.mli (and commons/Path.ml)

Reply:
In Path.mli I would also put a toplevel comment explaining the purpose of the module (a bit of what you had in your PR subject).
An important invariant is that because we internally use Unix.realpath, this means Path.t can be used only to represent existing file on the disk! it can't be used to give a name to a file you would like to create.

Reply:
Path_.ml

Reply:
There's already a set_.ml and map_.ml in commons/ so it will be in good company.
You can also do File_path.ml, given we have a file_type.ml

Reply:
FPath.ml